cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -751,19 +757,26 @@ handleExUnitsErrors ::
   -> Map ScriptWitnessIndex ScriptExecutionError
   -> Map ScriptWitnessIndex ExecutionUnits
   -> Either TxBodyErrorAutoBalance (Map ScriptWitnessIndex ExecutionUnits)
-handleExUnitsErrors scriptValidity failures exUnitsMap =
-    if null relevantFailures
+handleExUnitsErrors ScriptValid failuresMap exUnitsMap =
+    if null failures
       then Right exUnitsMap
-      else Left (TxBodyScriptExecutionError relevantFailures)
-  where relevantFailures :: [(ScriptWitnessIndex, ScriptExecutionError)]
-        relevantFailures = filter byScriptValidity (Map.toList failures)
-        byScriptValidity :: (ScriptWitnessIndex, ScriptExecutionError) -> Bool
-        byScriptValidity (_, e) = case scriptValidity of
-          ScriptValid -> True
-          ScriptInvalid -> case e of
-            ScriptErrorEvaluationFailed _ -> False
+      else Left (TxBodyScriptExecutionError failures)
+  where failures :: [(ScriptWitnessIndex, ScriptExecutionError)]
+        failures = Map.toList failuresMap
+handleExUnitsErrors ScriptInvalid failuresMap exUnitsMap
+  | null scriptFailures = Left TxBodyScriptBadScriptValidity
```

Comment:
We should plumb through the `ScriptWitnessIndex` with the error. Otherwise in the event where there are multiple scripts, the user will not know which script(s) are failing.

Reply:
In this case, we're erroring out because there are no script failures, so there aren't any indexes to plumb through.